documentation/modules/auxiliary/gather/redis_extractor.md

```diff
@@ -0,0 +1,80 @@
+## Vulnerable Application
+
+This module attaches to a Redis instance and extracts all stored keys and their associated data. If multiple databases are present the module will iterate through each.
+
+This module works on Redis versions 2.8.0 and later, and has been tested on versions through 6.0.8.
+
+To prepare a test instance of Redis,first install Redis v2.8.0 or greater. This can be done in docker with:
+
+`docker run -d -p 6379:6379 --name redis redis`
+
+Next, add some data to the database:
+
+`echo 'set key1 value1' | nc 127.0.0.1 6379 > /dev/null`(MacOS, may differ on Linux)
+
+Alternately, to run docker with a password:
+
+```bash
+docker run -d -p 6379:6379 --name redis redis --requirepass abcde
+echo 'auth abcde \n set key2 value2' | nc 127.0.0.1 6379 > /dev/null
+``` 
+
+
+## Verification Steps
+
+1. Install Redis and add data as described above.
+2. Start `msfconsole`
+3. Do: `use auxiliary/gather/redis_extractor`
+4. Do: `set rhosts [ip.of.redis.app]` 
+5. Do: `set PASSWORD [redis_password]` (optional)
+5. Do: `check` (optional)
+6. You will receive information about the Redis instalce.
+7. Do: `run`
+8. You will receive a screendump of the cached keys and contet.
+9. A CSV file with keys and content will be saved in your loot directory.
```

Comment:
`5` is duplicated here. You can use `1.` for each line to have the list items automatically implemented by any markdown renderer.